custom_components/mcp23017/binary_sensor.py

```diff
@@ -28,13 +29,17 @@
     cv.positive_int: cv.string,
 })
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+_CHIPS_SCHEMA = vol.Schema({
     vol.Required(CONF_PINS): _SENSORS_SCHEMA,
+    vol.Optional(CONF_I2C_ADDRESS, default=DEFAULT_I2C_ADDRESS): vol.Coerce(int),
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_CHIPS): [_CHIPS_SCHEMA],
     vol.Optional(CONF_INVERT_LOGIC, default=DEFAULT_INVERT_LOGIC): cv.boolean,
     vol.Optional(CONF_INTERRUPT): cv.positive_int,
     vol.Optional(CONF_PULL_MODE, default=DEFAULT_PULL_MODE): vol.In(['UP', 'DOWN']),
-    vol.Optional(CONF_I2C_ADDRESS, default=DEFAULT_I2C_ADDRESS):
-    vol.Coerce(int),
+    #vol.Coerce(int),
```

Comment:
Q: what is this for?

Reply:
This coerces the I2C address, which is passed as a hex value, into an integer. The mcp23017 module should work with either hex or int values but for whatever reason the hex value when defined in the configuration.yaml would throw an "address not found error" and coercing it into an int fixes it.